src/USER-MISC/fix_pafi.cpp

```diff
@@ -0,0 +1,798 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------
+   Contributing authors: Thomas Swinburne (CNRS & CINaM, Marseille, France)
+
+   Please cite the related publication:
+   T.D. Swinburne and M.-C. Marinica, Unsupervised calculation of free energy barriers in large crystalline systems, Physical Review Letters 2018
+------------------------------------------------------------------------- */
+
+
+#include "fix_pafi.h"
+#include <mpi.h>
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+#include "math_extra.h"
+#include "atom.h"
+#include "force.h"
+#include "update.h"
+#include "modify.h"
+#include "domain.h"
+#include "region.h"
+#include "respa.h"
+#include "comm.h"
+#include "input.h"
+#include "variable.h"
+#include "random_mars.h"
+#include "memory.h"
+#include "error.h"
+#include "group.h"
+#include "citeme.h"
+
+
+
+
+
+using namespace LAMMPS_NS;
+
+static const char cite_user_pafi_package[] =
+  "USER-PAFI package:\n\n"
+  "@article{SwinburneMarinica2018,\n"
+  "author={T. D. Swinburne and M. C. Marinica},\n"
+  "title={Unsupervised calculation of free energy barriers in large crystalline systems},\n"
+  "journal={Physical Review Letters},\n"
+  "volume={276},\n"
+  "number={1},\n"
+  "pages={154--165},\n"
+  "year={2018},\n"
+  "publisher={APS}\n"
+  "}\n\n";
+
+
+using namespace FixConst;
+
+enum{NONE,CONSTANT,EQUAL,ATOM};
+
+/* ---------------------------------------------------------------------- */
+
+FixPAFI::FixPAFI(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg), idregion(NULL), random(NULL)
```

Comment:
**all** class members that are pointers should be initialized to NULL here. This is to make certain, that when using `clear` or unfix/fix to delete and create classes, all pointers are always NULL when not allocated and thus can be safely deleted/destroyed in the destructor.